logging/net_appender.go

```diff
@@ -63,8 +63,11 @@ type NetAppender struct {
 	hostname     string
 	remoteWriter *remoteLogWriterGRPC
 
-	toLogMutex   sync.Mutex
-	toLog        []*commonpb.LogEntry
+	// toLogMutex guards toLog and toLogOverflowsSinceLastSync.
+	toLogMutex                  sync.Mutex
+	toLog                       []*commonpb.LogEntry
+	toLogOverflowsSinceLastSync int
```

Comment:
Smart idea to just count overflows instead of tying to search `toLog` for overlap with the batch just sent to app.